flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/assigners/MySqlSnapshotSplitAssigner.java

```diff
@@ -128,9 +129,33 @@ private MySqlSnapshotSplitAssigner(
         this.assignedSplits = assignedSplits;
         this.splitFinishedOffsets = splitFinishedOffsets;
         this.assignerFinished = assignerFinished;
+        this.assignerSuspended = false;
         this.remainingTables = new LinkedList<>(remainingTables);
         this.isRemainingTablesCheckpointed = isRemainingTablesCheckpointed;
         this.isTableIdCaseSensitive = isTableIdCaseSensitive;
+        this.pendingTables = null;
+    }
+
+    public void setAssignerSuspended(final boolean assignerSuspended) {
+        this.assignerSuspended = assignerSuspended;
+    }
+
+    public void setAssignerFinished(final boolean assignerFinished) {
+        this.assignerFinished = assignerFinished;
+    }
+
+    public void updateRemainingTables() {
+        this.remainingTables.addAll(pendingTables);
+    }
+
+    public void updateStatus() {
+        assert assignerFinished;
+        assert assignerSuspended;
+        assert remainingTables.isEmpty();
```

Comment:
please do not use `assert` for condition check, you can use `Preconditions.checkState()`